lib/classes/Swift/Transport/AbstractSmtpTransport.php

```diff
@@ -223,7 +214,34 @@ public function send(Swift_Mime_SimpleMessage $message, &$failedRecipients = nul
             $this->eventDispatcher->dispatchEvent($evt, 'sendPerformed');
         }
 
-        $message->generateId(); //Make sure a new Message ID is used
+        return $sent;
+    }
+
+    /**
+     * Send a copy of the given Message.
+     *
+     * The return value is the number of recipients who were accepted for delivery.
+     *
+     * @param string[] $failedRecipients An array of failures by-reference
+     *
+     * @return int
+     */
+    public function sendCopy(Swift_Mime_SimpleMessage $message, string $reversePath, array $tos, array $bcc, array &$failedRecipients = null)
+    {
+        $sent = 0;
+        $failedRecipients = (array) $failedRecipients;
+
+        $message->setBcc([]);
+
+        try {
+            $sent += $this->sendTo($message, $reversePath, $tos, $failedRecipients);
+            $sent += $this->sendBcc($message, $reversePath, $bcc, $failedRecipients);
+        } catch (Exception $e) {
+            $message->setBcc($bcc);
+            throw $e;
+        }
```

Comment:
Can we use a `finally` clause there to avoid the duplication of the `setBcc` call?

Reply:
Done, thanks.